crates/oxc_linter/src/rules/eslint/no_useless_call.rs

```diff
@@ -0,0 +1,156 @@
+use oxc_ast::{ast::{Argument, CallExpression, ChainElement, Expression, StaticMemberExpression}, match_member_expression, visit::walk::walk_expression, AstKind};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{ast_util::*, context::LintContext, rule::Rule, rules::jest::expect_expect, utils::is_same_reference, AstNode};
+
+#[derive(Debug, Default, Clone)]
+pub struct NoUselessCall;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// 
+    /// Disallow unnecessary calls to `.call()` and `.apply()`
+    ///
+    /// ### Why is this bad?
+    /// 
+    /// This rule is aimed to flag usage of Function.prototype.call() and Function.prototype.apply() that can be replaced with the normal function invocation.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// ```
+    NoUselessCall,
+    nursery, // TODO: change category to `correctness`, `suspicious`, `pedantic`, `perf`, `restriction`, or `style`
+             // See <https://oxc.rs/docs/contribute/linter.html#rule-category> for details
+);
+
+fn no_useless_call_diagnostic(span1: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Disallow the use of undeclared variables.")
+        .with_label(span1)
+}
+
+fn is_array_argument(_expr: Option<&Argument>) -> bool {
+    let Some(expr) = _expr else {
+        return false;
+    };
+
+    expr.is_array()
+}
+
+fn is_apply_member(call_expr: &CallExpression, member_expr: &StaticMemberExpression) -> bool {
+    (member_expr.property.name == "call" && call_expr.arguments.len() >= 1) || 
```

Comment:
use `is_method_call`

Reply:
It seem to not. The `call_expr.callee` should skip the chain expression